src/v/wasm/wasmtime.cc

```diff
@@ -613,46 +613,15 @@ class wasmtime_engine : public engine {
     ss::future<model::record_batch>
     invoke_transform(model::record_batch batch, transform_probe* p) {
         std::unique_ptr<transform_probe::hist_t::measurement> m;
-        model::record_batch_header header = batch.header().copy();
```

Comment:
question: Batch header copy isn't captured anywhere. Was that intentional?

Reply:
Good question, yes it was intentional to drop. We actually don't want to keep things like producer ID and base sequence here anyways we're not actually using impotency and we don't want to confuse rm_stm.

Reply:
Actually I need to set PID to -1 for that. Nice catch!